src/mudlet.cpp

```diff
@@ -3744,6 +3744,103 @@ void mudlet::startAutoLogin(const QStringList& cliProfiles)
     }
 }
 
+void mudlet::handleTelnetUri(const QString& uri)
+{
+    // Parse telnet:// URI format: telnet://host[:port]
+    if (!uri.startsWith(qsl("telnet://"))) {
+        return;
+    }
+    
+    QString connectionString = uri.mid(9); // Remove "telnet://"
+    QString host;
+    int port = 23; // Default telnet port
+    
+    // Parse host and optional port
+    int colonIndex = connectionString.indexOf(':');
+    if (colonIndex != -1) {
+        host = connectionString.left(colonIndex);
```

Comment:
I'd suggest a different name for this variable - given that we do have a `Host` class a lot of the codebase is littered with `pHost` and `host` as pointers and references to instances of that class. Perhaps `hostAddress` or `hosrUrl` would be appropriate? Thoughts @vadi2?